pulp_service/pulp_service/app/authorization.py

```diff
@@ -23,33 +25,50 @@ def has_permission(self, request, view):
         # Admins have all permissions
         if request.user.is_superuser:
             return True
+
+        domain_name = request.META['PATH_INFO'].removeprefix(settings.API_ROOT).split("/")[0]
+        user = request.user
+
+        # The user that created the domain has permission to access that domain
+        if DomainOrg.objects.filter(domain__name=domain_name, user=user).exists():
+            return True
+
+        # Check if user is creating a domain
+        if request.META['PATH_INFO'].endswith('/default/api/v3/domains/') and request.META['REQUEST_METHOD'] == 'POST':
+            action = "domain_create"
+        else:
+            action = "domain_operation"
+
         # Decode the identity header
+        header_present = False
```

Comment:
in this whole if-block I would move the "get header" cases to their own method or at least move them to their own if-block instead of mixing the "checking the header" part with the "is this request allowed" part.